commit 0dda4a844e63ccbcff1053fff65649dab0fd348f
Author: Christoph M. Becker <cmbecker69@gmx.de>
Date:   Wed Jan 8 11:52:24 2020 +0100

    Fix #79078: Hypothetical use-after-free in curl_multi_add_handle()
    
    To avoid this, we have to verify the handlers already in
    `curl_multi_add_handle()`, not only in `curl_multi_exec()`.

diff --git a/ext/curl/multi.c b/ext/curl/multi.c
index d10a03f741..af31e28adf 100644
--- a/ext/curl/multi.c
+++ b/ext/curl/multi.c
@@ -92,6 +92,8 @@ PHP_FUNCTION(curl_multi_add_handle)
 		RETURN_FALSE;
 	}
 
+	_php_curl_verify_handlers(ch, 1);
+
 	_php_curl_cleanup_handle(ch);
 
 	GC_ADDREF(Z_RES_P(z_ch));
